src/components/TopicMenu/Topics/LevelingUp.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from "react"
+import { TopicSection } from "../../../model/Character/TopicSection.class"
+import buildTable from "../../../util/buildTable"
+import SectionTitleDesc from "../../../util/SectionTitleDesc"
+import TopicSectionTable from "../../../util/TopicSectionTable"
+
+
+
+
+
+function LevelingUpPage(props: any) {
+
+    var tempSection = new TopicSection()
+
+    const [width, setWidth] = useState(window.innerWidth)
+    const tableChangeWidth = 360
+    
+    React.useEffect(() => {
+        window.addEventListener("resize", handleResize );
+        return () => window.removeEventListener("resize", handleResize)
+    })
+    function handleResize() {
+        setWidth(window.innerWidth)
+    }
+    function handleLevelUp(splitDesc: string[]) {
+       buildLevelUpDesc(splitDesc)
+       buildTable(splitDesc, tempSection, 3)
+    }
+    function buildLevelUpDesc(splitDesc: string[]) {
+        for(let i = 0; i < splitDesc.length; i++) {
+            if (i < 5) {
+                tempSection.desc.push(splitDesc[i].replaceAll('*',''))
+            }
+        }
+    }
+    
+    handleLevelUp(props.splitDesc)
```

Comment:
I would keep this towards the top of the non function code. Having it all the way at the bottom makes it easy to miss.

Reply:
and maybe even put the functions below the return?